httpreplay/httpreplay.go

```diff
@@ -57,6 +57,16 @@ func NewRecorder(filename string, initial []byte) (*Recorder, error) {
 	return &Recorder{proxy: p}, nil
 }
 
+// ScrubBody will replace all parts of the request body that match any of the
+// provided patterns with CLEARED, on both recording and replay.
+// Use ScrubBody when the body information is secret or may change from run to
+// run.
+//
+// Pattern is taken literally except for *, which matches any sequence of characters.
```

Comment:
I don't want to make your life harder and I know this works for your application, but this pattern semantics doesn't make a lot of sense for a body: any `*` will match to the end of the entire body, which is unlikely to be useful. 

I think this should take a regexp, or a list of them. (In which case "Scrub" might make sense to distinguish it from the others.)

Reply:
I am open to this. In fact, it's how I originally implemented it. Two points though (and the 2nd one is what made me switch it to this):

1. It won't necessarily match until the end, you can put pattern after it. For example, in my example the pattern is something like "<Blocklist><Latest>*</Latest></Blocklist>". A regexp would be more powerful though.
2. This value is serialized to the record file. These "pattern" strings serialize fine, but the `regexp.Regexp` type does not.

I guess it could just be passed as a string, and converted to a regular expression internally?

Reply:
Let's do regexps, passed and stored as strings. (`Regexp.String` will give you back the original string, according to its doc.)

Reply:
Done, and added tests.